### PR TITLE
Add clang-format config and use it to format extension/

### PR DESCRIPTION
**Pull request by dstockwell**

### CODE REVIEW

**Comment by smalls** [comment]
The format looks fine to me (and I'm happy to sign off if you think this is ready).

**Comment by dstockwell** [comment]
> The format looks fine to me (and I'm happy to sign off if you think this is ready).

Sure, let's land this.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,11 @@
+BasedOnStyle: Google
+AlignAfterOpenBracket: AlwaysBreak
```
Comment (smalls) [inline .clang-format]: @dstockwell I just approved, but I remembered a question. How'd you arrive at this list of exceptions?

Comment (dstockwell) [reply .clang-format]: That's a good question. They were sourced internally but removing them, the only one that has any affect is ReflowComments, so they are probably all part of the google config now. I'll reset to just `BasedOnStyle: Google`

**Pull request merged**